extensions/notebook/src/test/managePackages/managePackagesDialog.test.ts

```diff
@@ -110,7 +110,20 @@ describe('Manage Package Dialog', () => {
 		let packageManageProviders = new Map<string, IPackageManageProvider>();
 		packageManageProviders.set(LocalCondaPackageManageProvider.ProviderId, new LocalCondaPackageManageProvider(undefined));
 		let model = TypeMoq.Mock.ofInstance(new ManagePackagesDialogModel(undefined, packageManageProviders));
-		let dialog = TypeMoq.Mock.ofInstance(new ManagePackagesDialog(model.object));
+		let extensionContext: vscode.ExtensionContext = {
```

Comment:
Don't do this - as Aditya has found out this is going to keep breaking every time vscode updates their API. Just mock it out if you don't care about the actual functionality.

Reply:
I'll check but for interface I have to create a stub class to be able to mock right? this is similar to stub 

---
In reply to: [456162227](https://github.com/microsoft/azuredatastudio/pull/10918#discussion_r456162227) [](ancestors = 456162227)

Reply:
You should just be able to do `const mock = TypeMoq.Mock.ofType<vscode.ExtensionContext>();` and then just setup the mocking as needed for what you use (since it'll all be undefined to start with).

Reply:
But also - if you really wanted to make this a lot better I'd suggest exporting the extension context from the extension similar to what Maddy did lately for the treeViewDataProviders in the Notebook extension https://github.com/microsoft/azuredatastudio/pull/11348/files#diff-3726744dced36185069010dac2860acbR123

That way you get access to the actual context which is useful for stuff like extension path and whatnot

Reply:
that's what I did here right? not sure if you mean something else 

---
In reply to: [456565715](https://github.com/microsoft/azuredatastudio/pull/10918#discussion_r456565715) [](ancestors = 456565715)

Reply:
It looks fine now for fixing the initial thing I called out. My latest comment is just saying that giving tests access to the actual extensionContext from the extension is useful (instead of mocking it out). It's not strictly necessary but something to keep in mind if you find yourself needing to mock out stuff like extensionPath a lot.

Reply:
yeah for some reason I thought I cannot create mock for interface. I was wrong

---
In reply to: [459002110](https://github.com/microsoft/azuredatastudio/pull/10918#discussion_r459002110) [](ancestors = 459002110)